src/Util/EdgeConnectionUtilService.php

```diff
@@ -0,0 +1,279 @@
+<?php
+
+/**
+ * Copyright 2019 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 2 as published by the
+ * Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_edge\Util;
+
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Exception\ClientException;
+use GuzzleHttp\Exception\TransferException;
+use Symfony\Component\Console\Style\StyleInterface;
+
+/**
+ * Defines an interface for Edge connection classes.
+ */
+class EdgeConnectionUtilService implements EdgeConnectionUtilServiceInterface {
```

Comment:
How about calling this as `ApigeeEdgeManagementCliService` or something else that provides a more descriptive and also generic name? I add `Cli` because this service (class) should be used only by CLIs tools.

Reply:
good idea, done